android/app/src/main/java/com/criticalblue/auth/demo/auth/AuthRepo.java

```diff
@@ -61,45 +63,92 @@
-
-        connectionBuilder = new SignedConnectionBuilder(app.getAssets());
-
-        AppAuthConfiguration.Builder builder = new AppAuthConfiguration.Builder();
-        builder.setConnectionBuilder(connectionBuilder);
-        authService = new AuthorizationService(app, builder.build());
         authState = null;
         userInfoUrl = null;
         clientId = null;
         redirectUri = null;
         authScope = null;
+
+        authService = new AuthorizationService(app, ApproovCustomConnection());
+    }
+
+    /**
+     * All the calls made by AppAuth will be secured by using the Approov Managed Trust Roots
+     * feature to protect the TLS channel against MitM attacks.
+     *
+     * @link https://approov.io/docs/latest/approov-usage-documentation/#managed-trust-roots
+     *
+     * @return The AppAuthConfiguration with a HttpsURLConnection secured by Approov
+     */
+    private AppAuthConfiguration ApproovCustomConnection() {
+        return new AppAuthConfiguration.Builder()
+                .setConnectionBuilder(new ConnectionBuilder() {
+                    @NonNull
+                    @Override
+                    public HttpsURLConnection openConnection(@NonNull Uri uri) throws IOException {
+                        URL url = new URL(uri.toString());
+                        HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
+
+                        io.approov.service.httpsurlconn.ApproovService.addApproov(connection);
+
+                        return connection;
+                    }
+                })
+                .build();
     }
 
+    /**
+     * Approov allows to fetch secrets just in time for being used to make API calls, provided that
+     * the mobile app passes the attestation, thus no more hardcoded secrets on the app code.
+     *
+     * @link https://approov.io/docs/latest/approov-usage-documentation/#secure-strings
+     *
+     * @return The client secret retrieved from the Approov Cloud service
+     */
+    private String fetchJustInTimeClientSecret() {
+        String clientSecret = null;
+
+        try {
+            clientSecret = io.approov.service.httpsurlconn.ApproovService.fetchSecureString("client_secret", null);
+        } catch (ApproovRejectionException e) {
+            // failure due to the attestation being rejected, e.getARC() and e.getRejectionReasons() may be used to present information to the user
+            // (note e.getRejectionReasons() is only available if the feature is enabled, otherwise it is always an empty string)
+            failLogin(new AuthException("Failed to fetch the client_secret. ARC: " + e.getARC() + "REASON: " + e.getRejectionReasons()));
+        } catch (ApproovNetworkException e) {
+            // failure due to a potentially temporary networking issue, allow for a user initiated retry
+            failLogin(new AuthException("Failed to fetch the client_secret. Check the network connectivity and try again."));
```

Comment:
Again - if this appears to the user then it should just be "no network connectvity - please try again" ot something like that.

Reply:
I just kept it specific because the app is a demo targetting devs, but I understand that this may induce devs to use the same approach in production.

I will log the specifics and keep the message as you suggested.

Reply:
Solved in the last commit.